pkg/kp/healthmanager.go

```diff
@@ -0,0 +1,340 @@
+package kp
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/square/p2/Godeps/_workspace/src/github.com/Sirupsen/logrus"
+	"github.com/square/p2/Godeps/_workspace/src/github.com/hashicorp/consul/api"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util/stream"
+)
+
+type consulSimpleHealthManager struct {
+	store  consulStore
+	logger logging.Logger
+}
+
+type consulSimpleHealthUpdater struct {
+	store      consulStore
+	logger     logging.Logger
+	lastCheck  time.Time
+	lastStatus string
+}
+
+// newSimpleHealthManager creates a simple health manager that writes a service's health
+// status to Consul every 15 seconds or whenever its status changes.
+func (c consulStore) newSimpleHealthManager(
+	_node string,
+	logger logging.Logger,
+) HealthManager {
+	return &consulSimpleHealthManager{
+		store:  c,
+		logger: logger,
+	}
+}
+
+// NewUpdater implements the HealthManager interface.
+func (m *consulSimpleHealthManager) NewUpdater(_service string) HealthUpdater {
+	return &consulSimpleHealthUpdater{
+		store:  m.store,
+		logger: m.logger,
+	}
+}
+
+// Close implements the HealthManager interface. consulSimpleHealthManager has no
+// resources to release.
+func (m *consulSimpleHealthManager) Close() {}
+
+// PutHealth implements the HealthUpdater interface.
+func (u *consulSimpleHealthUpdater) PutHealth(health WatchResult) error {
+	if health.Status != u.lastStatus || time.Since(u.lastCheck) > TTL/4 {
```

Comment:
After considering this a bit more, I'd rather not touch the old implementation's timeouts in this PR. This kind of behavior is necessary for a watchdog service: you will always want to refresh the timeout well in advance of its expiration. It increases, not decreases, the write traffic by 4×.

The contract being advertised is:
> // newSimpleHealthManager creates a simple health manager that writes a service's health
// status to Consul every 15 seconds or whenever its status changes.